test/dynamics/backend/test_qiskit_experiments.py

```diff
@@ -0,0 +1,100 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""
+Integration tests that ensure this module interacts properly with qiskit-experiments.
+"""
+
+import numpy as np
+
+from qiskit import pulse
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import XGate, SXGate
+from qiskit.transpiler import Target
+from qiskit.providers.backend import QubitProperties
+from qiskit_experiments.calibration_management.calibrations import Calibrations
+from qiskit_experiments.library.calibration.rough_frequency import RoughFrequencyCal
+
+from qiskit_dynamics import Solver, DynamicsBackend
+from ..common import QiskitDynamicsTestCase, TestJaxBase
+
+
+class TestExperimentsIntegration(QiskitDynamicsTestCase, TestJaxBase):
+    """Test class for verifying correct integration with qiskit-experiments. Set to use JAX for
+    speed.
+    """
+
+    def setUp(self):
+        """Build simple simulator for multiple tests."""
+
+        # build the solver
+        solver = Solver(
+            static_hamiltonian=2 * np.pi * 5.0 * np.array([[-1.0, 0.0], [0.0, 1.0]]) / 2,
+            hamiltonian_operators=[2 * np.pi * np.array([[0.0, 1.0], [1.0, 0.0]]) / 2],
+            rotating_frame=2 * np.pi * 5.0 * np.array([[-1.0, 0.0], [0.0, 1.0]]) / 2,
+            hamiltonian_channels=["d0"],
+            channel_carrier_freqs={"d0": 5.0},
+            dt=1.0 / 4.5,
+        )
+
+        # build target gate definitions
+        target = Target()
+        target.qubit_properties = [QubitProperties(frequency=5.0)]
+
+        target.add_instruction(XGate())
+        target.add_instruction(SXGate())
+
+        self.simple_backend = DynamicsBackend(
+            solver=solver,
+            target=target,
+            solver_options={"method": "jax_odeint", "atol": 1e-6, "rtol": 1e-8},
+        )
+
+    def test_RoughFrequencyCal_calibration(self):
+        """Test RoughFrequencyCal outputs a sensible answer."""
+
+        cals = Calibrations()
+
+        dur = Parameter("dur")
+        amp = Parameter("amp")
+        sigma = Parameter("sigma")
+        beta = Parameter("beta")
+        drive = pulse.DriveChannel(Parameter("ch0"))
+
+        # Define and add template schedules.
+        with pulse.build(name="x") as x:
+            pulse.play(pulse.Drag(dur, amp, sigma, beta), drive)
+
+        with pulse.build(name="sx") as sx:
+            pulse.play(pulse.Drag(dur, Parameter("amp"), sigma, Parameter("beta")), drive)
```

Comment:
[Q] Why do you define variables "amp" and "beta" again ? Maybe I don't know how to use parameter.

Reply:
Good catch, I actually hadn't thought about this as I had just copied this code over from the qiskit-experiments [single qubit calibration tutorial](https://qiskit.org/documentation/experiments/tutorials/calibrating_real_device.html).

I just asked the dev team for experiments, and it was explained to me that it's necessary to define a new parameter so that the `X` and `SX` gate parameters are not linked. If you use the same `Parameter` instance in both, they will be linked, and it will mess with the tuning of each gate.

Reply:
I've just changed the test to remove the lines
```
amp = Parameter('amp')
beta = Parameter('beta')
```
and changed the definition of `x` to directly instantiate those parameters when the constructor is called, similar to `sx`. It saves a couple lines of code.